include/modem/location.h

```diff
@@ -374,9 +398,25 @@ int location_agps_data_process(const char *buf, size_t buf_len);
  *
  * @return 0 on success, or negative error code on failure.
  * @retval -EINVAL Given buffer is NULL or buffer length zero.
+ * @retval -ENOTSUP CONFIG_LOCATION_GNSS_PGPS_EXTERNAL is not set.
  */
 int location_pgps_data_process(const char *buf, size_t buf_len);
 
+/**
+ * @brief Pass cellular location back to library.
+ *
+ * @details If Location library is not receiving cellular position directly from services,
+ * it triggers the LOCATION_EVT_NEIGHBOR_CELL_REQUEST event when neighbor cell information
+ * should be sent to cloud services for location resolution. Once application has obtained
+ * the location it can be handed over to Location library using this function.
+ *
+ * @param[in] result Cellular location results. NULL indicates that location
+ *                   couldn't be retrieved.
+ *
+ * @return 0 on success, or negative error code on failure.
+ * @retval -ENOTSUP CONFIG_LOCATION_METHOD_CELLULAR_EXTERNAL is not set.
```

Comment:
This is aligned with `location_agps_data_process()` although that function has `-EINVAL` too for another error case.

Reply:
Yes, I noticed that. Still, it could be handled with an assert in the other functions as well.

BTW, `-EINVAL` is returned in case the provided buffer or the length of the buffer is zero. I think also that might be replaced by an assert, because these should only happen in case there's a bug in the calling code.

If the error isn't transient and the application can't recover from it, I don't think returning an error really helps. It just adds complexity both in the function itself and the caller.

Reply:
I know all this and don't disagree. Change agps and pgps functions so I'll align with those new implementations :)